packages/calypso-e2e/src/media-helper.ts

```diff
@@ -58,42 +69,53 @@ export function deleteFile( filePath: string ): void {
  * @param {string} param0.sourceFileName Basename of the source file to be cloned.
  * @param {string} [param0.testFileName] Basename of the test file to be generated.
  * @returns {string} Full path to the generated test file.
+ * @throws {Error} If file extension was not detected on source file.
  */
 export function createTestFile( {
 	sourceFileName,
 	testFileName,
 }: {
 	sourceFileName: string;
 	testFileName?: string;
-} ): string {
-	let fileName = getTimestamp();
+} ): TestFile {
+	let filename = getTimestamp();
 	// If the output `testFileName` is defined, use that as part of the final filename.
 	if ( testFileName ) {
-		fileName += `-${ testFileName }`;
+		filename += `-${ testFileName }`;
 	}
 
-	// Reassign the variable with the final name to be used, including the extension.
-	fileName = `${ fileName }.${ sourceFileName.split( '.' ).pop() }`;
-
-	const sourceFileDir = path.join( __dirname, '../../../../../test/e2e/image-uploads/' );
-	const sourceFilePath = path.join( sourceFileDir, sourceFileName );
+	const extension = sourceFileName.split( '.' ).pop();
+	if ( ! extension ) {
+		throw new Error( `Extension not found on source file ${ sourceFileName }` );
+	}
+	const basename = `${ filename }.${ sourceFileName.split( '.' ).pop() }`;
+	// Create test files in the same directory as the source file.
+	const dirname = path.join( __dirname, '../../../../../test/e2e/image-uploads/' );
+	// Full path on disk of the source file, to be copied and renamed.
+	const sourceFilePath = path.join( dirname, sourceFileName );
```

Comment:
Previous code did not survive the 3-month later check; I was confused by the reassignment of variables stemming from my attempts to limit the number of constants.

I've gone the other way now with this PR and creating single-use constants that better reflect the individual parts of the path, which can be assembled into an object implementing the interface at end.